src/main/java/net/countercraft/movecraft/combat/features/directors/AADirectors.java

```diff
@@ -95,44 +96,42 @@ private void processFireball(@NotNull SmallFireball fireball) {
         fireball.setShooter(p);
 
         Vector fireballVector = fireball.getVelocity();
-        double speed = fireballVector.length(); // store the speed to add it back in later, since all the values we will be using are "normalized", IE: have a speed of 1
+        double speed = fireballVector.length() * FireballSpeed; // store the speed to add it back in later, since all the values we will be using are "normalized", IE: have a speed of 1
         fireballVector = fireballVector.normalize(); // you normalize it for comparison with the new direction to see if we are trying to steer too far
 
         Block targetBlock = DirectorUtils.getDirectorBlock(p, AADirectorRange);
         Vector targetVector;
-        if (targetBlock == null || targetBlock.getType().equals(Material.AIR)) // the player is looking at nothing, shoot in that general direction
+
+        if (targetBlock == null || targetBlock.getType().isAir()) // the player is looking at nothing, shoot in that general direction
             targetVector = p.getLocation().getDirection();
         else { // shoot directly at the block the player is looking at (IE: with convergence)
             targetVector = targetBlock.getLocation().toVector().subtract(fireball.getLocation().toVector());
             targetVector = targetVector.normalize();
         }
 
-        if (targetVector.getX() - fireballVector.getX() > 0.5)
-            fireballVector.setX(fireballVector.getX() + 0.5);
-        else if (targetVector.getX() - fireballVector.getX() < -0.5)
-            fireballVector.setX(fireballVector.getX() - 0.5);
-        else
-            fireballVector.setX(targetVector.getX());
-
-        if (targetVector.getY() - fireballVector.getY() > 0.5)
-            fireballVector.setY(fireballVector.getY() + 0.5);
-        else if (targetVector.getY() - fireballVector.getY() < -0.5)
-            fireballVector.setY(fireballVector.getY() - 0.5);
-        else
-            fireballVector.setY(targetVector.getY());
-
-        if (targetVector.getZ() - fireballVector.getZ() > 0.5)
-            fireballVector.setZ(fireballVector.getZ() + 0.5);
-        else if (targetVector.getZ() - fireballVector.getZ() < -0.5)
-            fireballVector.setZ(fireballVector.getZ() - 0.5);
-        else
-            fireballVector.setZ(targetVector.getZ());
```

Comment:
Reverting these changes fixed directors for me.  I'm pretty sure I tried a change like the one you did here before, and it failed to work similarly.